doc/man3/EVP_PKEY_CTX_ctrl.pod

```diff
@@ -152,20 +254,58 @@ B<nid>. The B<nid> parameter must be B<NID_ffdhe2048>, B<NID_ffdhe3072>,
 B<NID_ffdhe4096>, B<NID_ffdhe6144> or B<NID_ffdhe8192>.  This macro can be
 called during parameter or key generation.
 
+The EVP_PKEY_CTX_set_dh_rfc5114() and EVP_PKEY_CTX_set_dhx_rfc5114() set the
+RFC5114 parameters instead of generating new DH parameters. Possible values are
+1, 2 and 3 corresponding to RFC5114 sections 2.1, 2.2 and 2.3. If set all other
+DH parameter options are overriden. 0 can be used to clear the stored value.
+
+The EVP_PKEY_CTX_set_dh_kdf_type() macro sets the key derivation function type
+to B<kdf> for DH key derivation. Possible values are B<EVP_PKEY_DH_KDF_NONE>
+and B<EVP_PKEY_DH_KDF_X9_42>.
+
+The EVP_PKEY_CTX_set0_dh_kdf_oid() macro sets the key derivation function
+object identifier to B<oid> for DH key derivation.
```

Comment:
Does this actually correspond to how OpenSSL uses the OID or is it something the user is supposed to implement himself?
Looking at the code, the usage of this parameter seems to be buried deep into the ASN1 encoding so I couldn't quite figure it out.

Reply:
This is used to construct a block of DER which is then hashed to create the shared secret. It has to be the same OID used on both sides otherwise the shared secret that is derived will be different. AFAIK there is no checking to confirm that the eventual CEK is actually for the algorithm defined by the OID. This is something the user is expected to get right, i.e. don't say you are using one algorithm in the OID, but then end up using a different one.

Reply:
How about just adding: "This OID should identify the algorithm to be used with the Content Encryption Key."